src/worker.cpp

```diff
@@ -1185,4 +1184,281 @@ void Worker::collectFrusFromJson()
         }
     }
 }
+
+std::string Worker::getPrimaryHwPath()
+{
+    if (m_parsedJson.contains("backupAndRestore"))
+    {
+        return m_parsedJson["backupAndRestore"].value("PrimaryHardwarePath",
+                                                      "");
+    }
+    return std::string{};
+}
+
+std::string Worker::getSecondaryHwPath()
+{
+    if (m_parsedJson.contains("backupAndRestore"))
+    {
+        return m_parsedJson["backupAndRestore"].value("SecondaryHardwarePath",
+                                                      "");
+    }
+    return std::string{};
+}
+
+types::BackupMapVariant Worker::getBackupMap()
+{
+    if (m_parsedJson.contains("backupAndRestore"))
+    {
+        const std::string& l_backupMap =
+            m_parsedJson["backupAndRestore"].value("BackupMap", "");
+        if (l_backupMap.find("IBMSystemBackupMap") != std::string::npos)
+        {
+            return utils::IBMSystemBackupMap;
+        }
+    }
+    return std::monostate{};
+}
+
+void Worker::backupAndRestore(types::VPDMapVariant& io_parsedVpdMap)
+{
+    logging::logMessage("BackupAndRestore is Started.");
+    const std::string& l_primaryHwPath = getPrimaryHwPath();
+    const std::string& l_secondaryHwPath = getSecondaryHwPath();
+
+    if (l_primaryHwPath.empty() && l_secondaryHwPath.empty())
+    {
+        logging::logMessage(
+            "Both primary and secondary path's are empty, can't initiate BackupAndRestore.");
+        return;
+    }
+
+    const auto& l_backupMap = getBackupMap();
+    if (std::holds_alternative<std::monostate>(l_backupMap))
+    {
+        logging::logMessage(
+            "Couldn't find system specific backup map, can't initiate BackupAndRestore.");
+        return;
+    }
+
+    types::VPDMapVariant l_parsedPrimaryVpdMap;
+    if (!l_primaryHwPath.empty())
+    {
+        fillVPDMap(l_primaryHwPath, l_parsedPrimaryVpdMap);
+    }
+
+    types::VPDMapVariant l_parsedSecVpdMap;
+    if (!l_secondaryHwPath.empty())
+    {
+        fillVPDMap(l_secondaryHwPath, l_parsedSecVpdMap);
+    }
+
+    if (!l_primaryHwPath.empty() && !l_secondaryHwPath.empty() &&
+        l_parsedPrimaryVpdMap.index() != l_parsedSecVpdMap.index())
+    {
+        logging::logMessage(
+            "Primary and secondary VPD types are not matching.");
+        return;
+    }
+
+    // Implement backup and restore for ipz type VPD
+    if (std::holds_alternative<types::IPZVpdMap>(l_parsedPrimaryVpdMap) ||
+        std::holds_alternative<types::IPZVpdMap>(l_parsedSecVpdMap))
+    {
+        if (const auto& l_backupInfoMap =
+                std::get_if<types::IpzBackupMap>(&l_backupMap))
+        {
+            auto l_primaryVpdMap = types::IPZVpdMap{};
+            if (!l_primaryHwPath.empty())
+            {
+                const auto l_ipzVpd =
```

Comment:
Error check for null?

Reply:
If parse fails for any reason, we get an exception.
so not checking for null .